frontend/src/pages/NewRunV2.tsx

```diff
@@ -288,11 +301,13 @@ function NewRunV2(props: NewRunV2Props) {
       description: runDescription,
       display_name: runName,
       experiment_id: apiExperiment?.id,
-      // pipeline_spec and pipeline_version_id is exclusive.
-      pipeline_spec: !(existingRun?.pipeline_version_id || existingPipelineVersion?.id)
+      // pipeline_spec and pipeline_version_reference is exclusive.
+      pipeline_spec: !(pipelineVersionRefClone || pipelineVersionRefNew)
         ? JsYaml.safeLoad(templateString || '')
         : undefined,
-      pipeline_version_id: existingRun?.pipeline_version_id || existingPipelineVersion?.id,
+      pipeline_version_reference: cloneOrigin.isClone
```

Comment:
Since pipeline_spec and pipeline_version_reference need to be exclusive, I think the condition check here for pipeline_version_reference should be: 

```
pipeline_version_reference: (pipelineVersionRefClone || pipelineVersionRefNew) ? 
              ( cloneOrigin.isClone ? pipelineVersionRefClone  : pipelineVersionRefNew ) : 
               undefined,
```

Is that right?